Pharmacy/PharamcyApi/Controllers/MedicineController.cs

```diff
@@ -26,13 +28,23 @@ public Medicine Add(Medicine medicine)
         [HttpGet]
         public IEnumerable<Medicine> GetAll()
         {
-            return _uow.GetRepository<IMedicineReadRepository>().GetAll();
+            return _uow.GetRepository<IMedicineReadRepository>().GetAll().Include(medicine => medicine.Manufacturer);
         }
 
         [HttpGet("{id}")]
         public Medicine GetById(int id)
         {
-            return _uow.GetRepository<IMedicineReadRepository>().GetById(id);
+            return _uow.GetRepository<IMedicineReadRepository>().GetAll()
+                .Include(medicine => medicine.Manufacturer)
+                .FirstOrDefault(medicine => medicine.Id == id);
+        }
+
+        [HttpGet("{name}")]
+        public Medicine GetByName(string name)
+        {
```

Comment:
Maybe would first check if we have it in the db, and if not, would return BadRequest rather than Ok but the data is null